google/detectors/credentials/ncrack/src/main/java/com/google/tsunami/plugins/detectors/credentials/ncrack/client/NcrackClient.java

```diff
@@ -213,6 +228,14 @@ public TargetService getTargetService() {
     return passwordList;
   }
 
+  public String getMaxTime() {
+    if (clioptions != null && clioptions.maxTime != null) {
```

Comment:
Instead of null checking here, you could consider passing in Optional<NcrackClientCliOptions> to the constructor, to make it clear that the config is optional. 

Also please add a unit test to make sure the config is optional.

Reply:
- The null checking has been replaced by an optional with a new commit.

- unit test has been added.